src/node_report.cc

```diff
@@ -1021,6 +1027,65 @@ const static struct {
     }
   }
 #endif
+
+  fprintf(fp, "\nLoaded Libraries\n");
+  PrintLoadedLibraries(fp, isolate);
+}
+
+/*******************************************************************************
+ * Function to print operating system information.
+ *
+ ******************************************************************************/
+#ifdef __linux__
+int
+LibraryPrintCallback(struct dl_phdr_info *info, size_t size, void *data) {
+  FILE* fp = (FILE*)data;
+  if (info->dlpi_name != nullptr && *info->dlpi_name != '\0') {
+    fprintf(fp, "  %s\n", info->dlpi_name);
+  }
+  return 0;
+}
+#endif
+
+static void PrintLoadedLibraries(FILE* fp, Isolate* isolate) {
+#ifdef __linux__
+  dl_iterate_phdr(LibraryPrintCallback, fp);
+#elif __APPLE__
+  int i = 0;
+  const char *name = _dyld_get_image_name(i);
+  while (name != nullptr) {
+    fprintf(fp, "  %s\n", name);
+    i++;
+    name = _dyld_get_image_name(i);
+  }
+#elif _AIX
+  const unsigned int buffer_inc = 4096;
+  unsigned int buffer_size = buffer_inc;
+  char* buffer = (char*) malloc(buffer_size);
+  int rc = loadquery(L_GETINFO, buffer, buffer_size);
+  while (rc != 0 && buffer_size < 1024 * 1024 && buffer != nullptr) {
+    free(buffer);
+    buffer_size += buffer_inc;
+    buffer = (char*) malloc(buffer_size);
+    rc = loadquery(L_GETINFO, buffer, buffer_size);
+  }
+  if (rc == 0 && buffer != nullptr) {
+    char* buf = buffer;
+    ld_info* cur_info = (ld_info*) buf;
+    do {
+      char* member_name = cur_info->ldinfo_filename
+        + strlen(cur_info->ldinfo_filename) + 1;
+      if (*member_name != '\0') {
+        fprintf(fp, "  %s(%s)\n", cur_info->ldinfo_filename, member_name);
+      } else {
+        fprintf(fp, "  %s\n", cur_info->ldinfo_filename);
+      }
+      buf += cur_info->ldinfo_next;
+      cur_info = (ld_info*) buf;
+    } while (cur_info->ldinfo_next != 0);
```

Comment:
I think this will skip over the last entry (i.e. in the previous line cur_info is updated to the last entry, so now `cur_info->ldinfo_next == 0`, but cur_info hasn't yet been processed).